src/hooks/useCurrency.ts

```diff
@@ -76,23 +77,29 @@ export function useTokenFromNetwork(tokenAddress: string | null | undefined): To
  * Returns null if token is loading or null was passed.
  * Returns undefined if tokenAddress is invalid or token does not exist.
  */
-export function useTokenFromMapOrNetwork(tokens: TokenMap, tokenAddress?: string | null): Token | null | undefined {
+export function useTokenFromMapOrNetwork(
+  tokens: TokenMap,
+  tokenAddress?: string | null,
+  skipNetwork = false
+): Token | null | undefined {
   const address = isAddress(tokenAddress)
   const token: Token | undefined = address ? tokens[address] : undefined
 
   const tokenFromNetwork = useTokenFromNetwork(token ? undefined : address ? address : undefined)
 
-  return tokenFromNetwork ?? token
+  return skipNetwork || !tokenFromNetwork ? token : tokenFromNetwork
```

Comment:
If `skipNetwork` you should return `undefined`:

    return skipNetwork ? undefined : tokenFromNetwork || token